.ci/ci-library.sh

```diff
@@ -114,29 +73,18 @@ execute(){
     cd - > /dev/null
 }
 
-# Sort packages by dependency
-define_build_order() {
-    local sorted_packages=()
-    for unsorted_package in "${packages[@]}"; do
-        _build_add "${unsorted_package}"
-    done
-    packages=("${sorted_packages[@]}")
-}
-
 # Added commits
 list_commits()  {
     _list_changes commits '*' '#*::' --pretty=format:'%ai::[%h] %s'
 }
 
-# Changed recipes
+# Get changed packages in correct build order
 list_packages() {
-    local _packages
-    _list_changes _packages '*/PKGBUILD' '%/PKGBUILD' --pretty=format: --name-only || return 1
-    for _package in "${_packages[@]}"; do
-        local find_case_sensitive="$(find -name "${_package}" -type d -print -quit)"
-        test -n "${find_case_sensitive}" && packages+=("${_package}")
-    done
-    return 0
+    _as_list "packages" "*" "" "$($DIR/ci-get-build-order.py)"
```

Comment:
One thing I didn't realise when switching to use `_as_list`, it assumes that no output is a failure, but in this case it should actually be allowed (see line 27 in ci-build.sh) - that's why the build is failing right now.

Reply:
msys2/MINGW-packages#9545